backup-stores/azure/src/main/java/io/camunda/zeebe/backup/azure/ManifestManager.java

```diff
@@ -0,0 +1,194 @@
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import io.camunda.zeebe.backup.api.Backup;
+import io.camunda.zeebe.backup.api.BackupIdentifier;
+import io.camunda.zeebe.backup.azure.AzureBackupStoreException.UnexpectedManifestState;
+import io.camunda.zeebe.backup.azure.manifest.Manifest;
+import io.camunda.zeebe.backup.azure.manifest.Manifest.InProgressManifest;
+import io.camunda.zeebe.backup.azure.manifest.Manifest.StatusCode;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+
+public final class ManifestManager {
+  public static final int PRECONDITION_FAILED = 412;
+
+  /**
+   * The path format consists of the following elements:
+   *
+   * <ul>
+   *   <li>{@code "manifests"}
+   *   <li>{@code partitionId}
+   *   <li>{@code checkpointId}
+   *   <li>{@code nodeId}
+   *   <li>{@code "manifest.json"}
+   * </ul>
+   *
+   * The path format is constructed by partitionId/checkpointId/nodeId/manifest.json
+   */
+  private static final String MANIFEST_PATH_FORMAT = "%s/%s/%s/manifest.json";
+
+  private static final ObjectMapper MAPPER =
+      new ObjectMapper()
+          .registerModule(new Jdk8Module())
+          .registerModule(new JavaTimeModule())
+          .disable(WRITE_DATES_AS_TIMESTAMPS)
+          .setSerializationInclusion(Include.NON_ABSENT);
+  private boolean containerCreated = false;
+  private final BlobContainerClient blobContainerClient;
+
+  ManifestManager(final BlobContainerClient blobContainerClient) {
+    this.blobContainerClient = blobContainerClient;
+  }
+
+  PersistedManifest createInitialManifest(final Backup backup) {
+
+    final var manifest = Manifest.createInProgress(backup);
+    final byte[] serializedManifest;
+    assureContainerCreated();
+    try {
+      serializedManifest = MAPPER.writeValueAsBytes(manifest);
+    } catch (final JsonProcessingException e) {
+      throw new RuntimeException(e);
+    }
+    final BlobClient blobClient = blobContainerClient.getBlobClient(manifestPath((manifest)));
+
+    try {
+      final BlobRequestConditions blobRequestConditions = new BlobRequestConditions();
+      disableOverwrite(blobRequestConditions);
+      final Response<BlockBlobItem> blockBlobItemResponse =
+          blobClient.uploadWithResponse(
+              new BlobParallelUploadOptions(BinaryData.fromBytes(serializedManifest))
+                  .setRequestConditions(blobRequestConditions),
+              null,
+              Context.NONE);
+
+      return new PersistedManifest(blockBlobItemResponse.getValue().getETag(), manifest);
+    } catch (final BlobStorageException e) {
+      if (e.getErrorCode() == BlobErrorCode.BLOB_ALREADY_EXISTS) {
+        throw new UnexpectedManifestState(e.getMessage());
```

Comment:
:wrench: Let's provide a clear message such as "Manifest already exists" and propagate the full exception as cause. This should be a bit more helpful than just using Azure's error message.